nes-physical-operators/src/Aggregation/AggregationBuildPhysicalOperator.cpp

```diff
@@ -77,16 +77,50 @@ Interface::HashMap* getAggHashMapProxy(
     return aggregationSlice->getHashMapPtrOrCreate(workerThreadId);
 }
 
+void AggregationBuildPhysicalOperator::setup(ExecutionContext& executionCtx, CompilationContext& compilationContext) const
+{
+    WindowBuildPhysicalOperator::setup(executionCtx, compilationContext);
+
+    /// Creating the cleanup function for the slice of current stream
+    /// As the setup function does not get traced, we do not need to have any nautilus::invoke calls to jump to the C++ runtime
+    /// We are not allowed to use const or const references for the lambda function params, as nautilus does not support this in the registerFunction method.
+    /// ReSharper disable once CppPassValueParameterByConstReference
+    /// NOLINTBEGIN(performance-unnecessary-value-param)
+    const auto operatorHandler = dynamic_cast<AggregationOperatorHandler*>(executionCtx.getGlobalOperatorHandler(operatorHandlerId).value);
+    operatorHandler->cleanupStateNautilusFunction
+        = std::make_shared<CreateNewHashMapSliceArgs::NautilusCleanupExec>(compilationContext.registerFunction(std::function(
+            [copyOfFieldKeys = hashMapOptions.fieldKeys,
+             copyOfFieldValues = hashMapOptions.fieldValues,
+             copyOfEntriesPerPage = hashMapOptions.entriesPerPage,
+             copyOfEntrySize = hashMapOptions.entrySize,
+             copyOfAggregationFunctions = aggregationPhysicalFunctions](nautilus::val<Nautilus::Interface::HashMap*> hashMap)
```

Comment:
I am not sure is there a reason why this lambda capture is so verbose?

```suggestion
        = std::make_shared<CreateNewHashMapSliceArgs::NautilusCleanupExec>(compilationContext.registerFunction(std::function(
            [hashMapOptions, aggregationPhysicalFunctions](nautilus::val<Nautilus::Interface::HashMap*> hashMap)
```